Kyle

emeraldarcher opened the pull request:
Roll Controller finished

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
Roll Controller

Added ability to roll dice to Roll Controller

emeraldarcher committed:
Turn

Work on Turn Controller

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
Turn Tracker Controller

More work...

emeraldarcher committed:
Roll Controller

Cleanup

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
Roll Controller

Finished the RollController

CaptBoscho commented on java/src/client/roll/RollController.java:
```diff
@@ -19,6 +21,7 @@
 	public RollController(IRollView view, IRollResultView resultView) {
 		super(view);
 		setResultView(resultView);
+		roller = new Dice(2,12);
```
if you do it just like this, then the percentages of numbers rolled will be more equal than 2 die... like 2 has the same probability of being rolled as a 7. I would recommend making two die from 1-6 and adding up their rolls. What do you think?

CaptBoscho commented on java/src/client/turntracker/TurnTrackerController.java:
```diff
@@ -21,12 +33,30 @@ public ITurnTrackerView getView() {
 	@Override
 	public void endTurn() {
-
+//		facade.endTurn(); // TODO: 2/16/2016 Have Corbin implement in facade
```
It is finishTurn(int playerID). Requires the playerID just as a check.

CaptBoscho commented on java/src/client/turntracker/TurnTrackerControllerState.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public void endTurn() {
+//		facade.endTurn(); // TODO: 2/16/2016 Have Corbin implement in facade
```
:)

emeraldarcher committed:
Fixed dice

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
Dice

Fixed the Dice class - Wasn't initializing the list of die

derekargueta commented:
Currently don't have a way to test this yet. Are there any other game components that depend on this being merged into master ASAP?

emeraldarcher committed:
More work on the TurnTracker Controller

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
TurnTracker Controller

Lots of work...

emeraldarcher committed:
TurnTracker Controller

Finished minus testing and the facade needs to grant access to the
player manager or some method of getting all the player
info...supposedly this has been done, maybe it's not committed.

emeraldarcher committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Kyle

emeraldarcher committed:
TurnTracker Controller & dependents

Facade - Updated the getPlayers() method
ModelPlayerInfo - Added member and method for player index (id)
States - Clean up
TurnTrackerControllerState - Moved some stuff to the controller
IGame/Game - Misc. cleanup

emeraldarcher commented:
Is it possible to pick and choose what to merge? I have some changes outside of the controllers I'm working on that should be merged.

emeraldarcher committed:
PointsController & dependencies

PointsController - Implemented
PointsController States added/implemented

Facade - Added method for getting the winner
IGame/Game - Added method for getting the winner
IPlayerManager/PlayerManager - Added method for getting the winner
TurnTracker - Added phase for game finished
